tutorials/preprocessing/20_spatial_filter.py

```diff
@@ -0,0 +1,94 @@
+"""
+Spatial Filter
+==============
+
+This example demonstrates the effect of spatial filter on EEG data.
+"""
+
+#%%
+# .. include:: ../../../../links.inc
+
+#%%
+# .. note::
+#
+#     The lemon datasets used in this tutorial is composed of EEGLAB files. To
+#     use the MNE reader :func:`mne.io.read_raw_eeglab`, the ``pymatreader``
+#     optional dependency is required. Use the following installation method
+#     appropriate for your environment:
+#
+#     - ``pip install pymatreader``
+#     - ``conda install -c conda-forge pymatreader``
+#
+#     Note that an environment created via the `MNE installers`_ includes
+#     ``pymatreader`` by default.
+
+import matplotlib.pyplot as plt
+import mne
+import numpy as np
+from mne.io import read_raw_eeglab
+from mne.channels import find_ch_adjacency
+from mne.viz import plot_ch_adjacency
+
+from pycrostates.datasets import lemon
+from pycrostates.preprocessing import apply_spatial_filter
+
+raw_fname = lemon.data_path(subject_id='010004', condition='EC')
+raw = read_raw_eeglab(raw_fname, preload=True)
+
+raw.pick('eeg')
+raw.set_eeg_reference('average')
+
+#%%
+# Spatial filters were first introduced in the context of EEG source imaging
+# in\ :footcite:t:`michel2019eeg`, with the aim to reduce the impact of
+# local transient artifacts on EEG channels. The computation of a spatial
+# filter starts with the creation of an adjacency matrix, defining whether
+# pairs of electrodes are adjacent (neighbors).
+# An adjacency matrix can be represented as a graph where each node represents
+# a sensor and each edge links adjacent sensors.
+
+adjacency, ch_names = find_ch_adjacency(info=raw.info, ch_type="eeg")
+plot_ch_adjacency(raw.info, adjacency, ch_names, kind="2d", edit=False)
+plt.show()
```

Comment:
I like the explanation of the internals, especially as the computation of an adjacency matrix can be erroneous. `plot_ch_adjacency` has an interactive component to edit the adjacency matrix. I think it would be good to have an additional argument to the spatial filter function:

```
from scipy.sparse import csr_matrix

def [...](ch_adjacency: Union[csr_matrix, str] = "auto"):
```

And to edit accordingly the error checking, and this tutorial to include a note about the possible editing of the adjacency matrix. For instance, in this tutorial, I would remove the link between `Cz` and `Fz`; between `Fp1` and `Fp2`, between `PO9` and `PO10`, ... yielding:

![image](https://user-images.githubusercontent.com/73893616/213926183-ca48dc41-b60d-4313-a326-7ecc13f302c2.png)